### PR TITLE
[master <  T0814-MG] Fix CentOS 7 build by top-level memgraph namespace

### PR DESCRIPTION
**Pull request by jbajic**
The issue is with the openssl version API compactibility, which was solved by using boost socket. That caused an issue with our `query` namespace and that's why we are making a huge refactoring.
[master < Task] PR
- [ ] Check, and update documentation if necessary
- [ ] Update [changelog](https://docs.memgraph.com/memgraph/changelog)

### CODE REVIEW

File: tests/benchmark/query/execution.cpp
Comment (gitbuda) [inline tests/benchmark/query/execution.cpp:39]: Since this is a cpp file is it possible to put all these under `memgraph` namespace, to avoid repetition of `memgraph::`? @jbajic if that's a hustle feel free to ignore it...

**Review by gitbuda — COMMENTED** [review]
A bunch of heavy lifting here 💪 Very nice!

In addition:
* I would rename this PR into something like: "Fix CentOS 7 build by top-level memgraph namespace" (it's a bit longer but tells about the whole scope).
* Make a page under `Memgraph Internals` explaining why the namespace is important and how to deal with the namespaces correctly (if there is anything specific that contributors have to know).

File: .githooks/pre-commit
Comment (jbajic) [inline .githooks/pre-commit:41]: The reasoning behind this addition is that `header-checker` stops at the first header, but that's really inconvenient when dealing with refactoring tasks, so it will not stop, but it will remember that it has failed and it will return the end result as a failure. The behaviour is the same as before, no commit will happen but all files will have their license year updated.

**Review by kostasrim — COMMENTED** [review]
@jbajic I am very sorry to point this now but maybe you can `sed` the change quickly. `memgraph` is a very big name to warp up the currently existing namespaces and I would suggest renaming it to `mg`. This will solve the problem we have with centos without introducing the verbosity of the full name.

**Comment by jbajic** [comment]
> @jbajic I am very sorry to point this now but maybe you can `sed` the change quickly. `memgraph` is a very big name to warp up the currently existing namespaces and I would suggest renaming it to `mg`. This will solve the problem we have with centos without introducing the verbosity of the full name.

After discussing internally we have decided to stick with `memgraph` namespace. You can read more about it here => https://www.notion.so/memgraph/C-Code-Conventions-7497766e0e1647b987733b9e94171423#4f614c6605144610ab9512ac6c15c821

File: src/lisp/lcp.lisp
Comment (antaljanosbenjamin) [inline src/lisp/lcp.lisp:72]: I think these two are not necessary.

File: src/storage/v2/replication/serialization.cpp
Comment (antaljanosbenjamin) [inline src/storage/v2/replication/serialization.cpp:20]: Noy necessary.

File: src/storage/v2/replication/serialization.hpp
Comment (antaljanosbenjamin) [inline src/storage/v2/replication/serialization.hpp:26]: Not necessary.

File: src/storage/v2/replication/slk.cpp
Comment (antaljanosbenjamin) [inline src/storage/v2/replication/slk.cpp:22]: Not necessary.

File: src/storage/v2/replication/slk.hpp
Comment (antaljanosbenjamin) [inline src/storage/v2/replication/slk.hpp:22]: Not necessary.

File: src/storage/v2/edge_accessor.hpp
Comment (antaljanosbenjamin) [inline src/storage/v2/edge_accessor.hpp:111]: It is not something this PR should fix, just a question: doesn't this will result a very poor distribution of edge accessors in hashtables? As the Gid is strictly incremental, the set of edge accessors might get very close hashes. @gitbuda do you know any reasoning behind this?

File: src/utils/license.cpp
Comment (antaljanosbenjamin) [inline src/utils/license.cpp:252]: Not necessary.

File: src/utils/small_vector.hpp
Comment (antaljanosbenjamin) [inline src/utils/small_vector.hpp:70]: the closing and opening of `memgraph::utils` can be removed here I think.

**Review by antaljanosbenjamin — CHANGES_REQUESTED** [review]
Nice work Jure! Where I commented "Not necessary" there please check the whole file for extra `memgraph::` qualifiers. If you think I asked too much, let me know and I can jump on fixing them.

File: src/glue/auth.cpp
Comment (jbajic) [reply src/glue/auth.cpp:18]: You are right, I will remove them

File: src/io/network/serialization.hpp
Comment (jbajic) [reply src/io/network/serialization.hpp:20]: Aha yes the `slk` stuff, I had a lot of issues here that's why I've placed `memgraph::` everywhere I could, I'll reiterate through here

File: src/kvstore/kvstore_dummy.cpp
Comment (jbajic) [reply src/kvstore/kvstore_dummy.cpp:1]: To the trash, it goes

File: src/glue/auth.hpp
Comment (antaljanosbenjamin) [inline src/glue/auth.hpp:18]:
```suggestion
 * This function converts query::AuthQuery::Privilege to its corresponding
```

File: src/query/discard_value_stream.hpp
Comment (antaljanosbenjamin) [reply src/query/discard_value_stream.hpp:20]: This is not fixed.

**Review by antaljanosbenjamin — CHANGES_REQUESTED** [review]
Two small fixes and it is good to go!

File: src/storage/v2/edge_accessor.hpp
Comment (gitbuda) [reply src/storage/v2/edge_accessor.hpp:111]: Not sure, but I think nobody considered that (maybe somebody did), but since `gid` is the only data under edge that will exist for sure, doing hash like this is the most straightforward thing.

Comment (antaljanosbenjamin) [reply src/storage/v2/edge_accessor.hpp:111]: @kostasrim checked out, and it works just fine.

**Review by antaljanosbenjamin — APPROVED** [review]

**Review by kostasrim — APPROVED** [review]

**Comment by antaljanosbenjamin** [comment]
I compared the changes before and after the force-push and they were the same, so I don't wait for another review. I will also merge it without squashing to separate the changes of adding new namespace and introducing boost.

**Pull request merged**
